### PR TITLE
🧪 Scout: Add comprehensive CSV formula injection escaping tests

### PR DESCRIPTION
**Pull request by cungminh2710**
💡 What: Added comprehensive behavior-driven unit tests for CSV formula injection escaping in `internal/csvsafe/csvsafe_scout_test.go`.

🎯 Why: Protect CSV localized data export and security filters against spreadsheet injection vectors (such as leading control characters, placement-specific escaping rules) and verify row boundary conditions.

🧩 Scope: `internal/csvsafe/csvsafe_scout_test.go`, `.jules/scout.md`

✅ Verification: Ran `go test -v ./internal/csvsafe/...` and workspace-wide `go test ./...`.

🛡️ Confidence: Prevents regression in formula injection escaping when localized CSV data is processed.

---
*PR created automatically by Jules for task [3559230340246819293](https://jules.google.com/task/3559230340246819293) started by @cungminh2710*

### CODE REVIEW

File: internal/csvsafe/csvsafe_scout_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+func TestEscapeRow_ScoutEdgeCases(t *testing.T) {
+	t.Run("nil row", func(t *testing.T) {
+		got := EscapeRow(nil)
+		if got != nil {
+			t.Fatalf("EscapeRow(nil) = %v, want nil", got)
+		}
+	})
+
+	t.Run("empty row", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply internal/csvsafe/csvsafe_scout_test.go:98]: @jules fix
